src/Domain/Collector.php

```diff
@@ -219,13 +219,12 @@ final class Collector
     /**
      * Creates a new instance of the Collector.
      *
-     * @param string $dir
-     * @param array<string> $files
+     * @param array<string> $directories
      */
-    public function __construct(string $dir, array $files)
+    public function __construct(array $directories)
     {
-        $this->dir = $dir;
-        $this->commonPath = $files !== [] ? PathShortener::extractCommonPath($files) : '';
+        $this->analysedDirectories = $directories;
+        $this->commonPath = PathShortener::extractCommonPath($directories);
```

Comment:
The `extractCommonPath` has be already done in `ConfigResolver`. We should use Configuration to retrieve it to avoid twice call ?

Reply:
Done